### PR TITLE
The plugins are also deployed when 'Sealer join' is executed

### PR DESCRIPTION
**Pull request by bxy4543**
The plugins are also deployed when 'Sealer join' is executed

### CODE REVIEW

File: apply/processor/scale.go [hunk 1]
```diff
@@ -61,51 +72,131 @@ func (s ScaleProcessor) Execute(cluster *v2.Cluster) error {
...
+
+func (s *ScaleProcessor) ScaleUpPipeLine() ([]func(cluster *v2.Cluster) error, error) {
+	var todoList []func(cluster *v2.Cluster) error
+	todoList = append(todoList,
+		s.GetPhasePluginFunc(plugin.PhaseOriginally),
+		s.RunConfig,
+		s.MountRootfs,
+		s.GetPhasePluginFunc(plugin.PhasePreInit),
+		s.Join,
+		s.GetPhasePluginFunc(plugin.PhasePreGuest),
```
Comment (fanux) [inline apply/processor/scale.go:111]: Join should not exec preguest plugins.

**Comment by fanux** [comment]
Maybe we need PreJoin phase, and only exec PreJoin plugins when join master or nodes.

Support multi phase:
```
action: PreJoin | PreInit
```

**Comment by bxy4543** [comment]
> Maybe we need PreJoin phase, and only exec PreJoin plugins when join master or nodes.
> 
> Support multi phase:
> 
> ```
> action: PreJoin | PreInit
> ```

great

File: apply/processor/scale.go [hunk 2]
```diff
@@ -61,51 +72,129 @@ func (s ScaleProcessor) Execute(cluster *v2.Cluster) error {
...
+
+func (s *ScaleProcessor) ScaleUpPipeLine() ([]func(cluster *v2.Cluster) error, error) {
+	var todoList []func(cluster *v2.Cluster) error
+	todoList = append(todoList,
+		s.GetPhasePluginFunc(plugin.PhaseOriginally),
+		s.RunConfig,
+		s.MountRootfs,
+		s.GetPhasePluginFunc(plugin.PhasePreJoin),
+		s.Join,
+		s.GetPhasePluginFunc(plugin.PhasePostInstall),
```
Comment (fanux) [inline apply/processor/scale.go:111]: I think only exec PhasePreJoin and PhasePostJoin

**Review by fanux — APPROVED** [review]

**Pull request merged**